x/mongo/driver/wiremessage/wiremessage.go

```diff
@@ -381,16 +381,9 @@ func ReadMsgSectionSingleDocument(src []byte) (doc bsoncore.Document, rem []byte
 // ReadMsgSectionDocumentSequence reads an identifier and document sequence from src and returns the document sequence
 // data parsed into a slice of BSON documents.
 func ReadMsgSectionDocumentSequence(src []byte) (identifier string, docs []bsoncore.Document, rem []byte, ok bool) {
-	length, rem, ok := readi32(src)
-	if !ok || int(length) > len(src) {
-		return "", nil, rem, false
-	}
-
-	rem, ret := rem[:length-4], rem[length-4:] // reslice so we can just iterate a loop later
-
-	identifier, rem, ok = readcstring(rem)
+	identifier, rem, ret, ok := ReadMsgSectionRawDocumentSequence(src)
 	if !ok {
-		return "", nil, rem, false
+		return "", nil, ret, false
```

Comment:
Why return `ret` here instead of `rem`? Should we always return either `src` or `[]byte{}` in the `!ok` condition instead?

Reply:
The `ret` returned by `ReadMsgSectionRawDocumentSequence` is the `rem` in the original logic. I just kept the returned variable names of `ReadMsgSectionRawDocumentSequence` to avoid confusion there. However, you are right that we only need to return an empty slice or even nil on false cases including those underneath.